source/MaterialXRuntime/Private/PvtPrim.cpp

```diff
@@ -82,6 +79,9 @@ void PvtPrim::destroy()
     }
     _primOrder.clear();
     _primMap.clear();
+
+    // Tag as disposed.
+    dispose(true);
```

Comment:
Having this before was causing all kinds of random exceptions to be thrown erroneously.